File: Leetcode/2410 - Maximum Matching of Players With Trainers/solution.py
def matchPlayersAndTrainers(self, players: List[int], trainers: List[int]) -> int:
    players.sort()
    trainers.sort()

    p: int = 0
    t: int = 0
    matches: int = 0

    while p < len(players) and t < len(trainers):
        if players[p] <= trainers[t]:
            matches += 1
            p += 1
        t += 1

    return matches

File: Leetcode/2410 - Maximum Matching of Players With Trainers/solution.cpp
int matchPlayersAndTrainers(vector<int> &players, vector<int> &trainers)
{
    sort(players.begin(), players.end());
    sort(trainers.begin(), trainers.end());
    int pItr = 0, tItr = 0;
    int matches = 0;
    while (pItr < players.size() && tItr < trainers.size())
    {
        if (players[pItr] <= trainers[tItr])
        {
            matches++;
            pItr++;
        }
        tItr++;
    }
    return matches;
}